clang/utils/TableGen/SveEmitter.cpp

```diff
@@ -1802,9 +1854,29 @@ void SVEEmitter::createStreamingAttrs(raw_ostream &OS, ACLEKind Kind) {
 
     if (Def->isFlagSet(IsStreamingFlag))
       StreamingMap["ArmStreaming"].insert(Def->getMangledName());
-    else if (Def->isFlagSet(VerifyRuntimeMode))
+    else if (Def->isFlagSet(VerifyRuntimeMode)) {
+      // Verify that the target guards contain at least one feature that
+      // actually enables SVE or SME (explicitly, or implicitly). This is needed
+      // for the code in SemaARM.cpp (checkArmStreamingBuiltin) that checks
+      // whether the required runtime mode for an intrinsic matches with the
+      // given set of target features and function attributes.
+      //
+      // The feature lists below must match the disabled features in
+      // 'checkArmStreamingBuiltin'!
+      if (!Def->getSVEGuard().empty() &&
+          !verifyGuard(Def->getSVEGuard(),
+                       {"sve", "sve2", "sve2p1", "sve2-aes", "sve2-sha3",
```

Comment:
I think we should need to have `"f32mm"` and  `"f64mm"` here (and the `CallerFeatureMapWithoutSVE` part). This works at the moment as the only intrinsics that have these in their target-guard also redundantly include `sve` and are invalid in streaming mode.

Reply:
Thanks for pointing those out!

It's worth saying that we could also implement things differently and require all SVE target guards to have "sve/sve2/sve2p1" as a base, such that `let SVETargetGuard = "sve2-aes"` becomes `let SVETargetGuard = "sve2,sve2-aes"`. That means we need to refactor some of the target guards in the .td files, but it means we don't have to continually add all features that imply sve/sve2/sve2p1 to this list (and the list in SemaARM.cpp). For end-users I think it doesn't matter, the only difference is in the diagnostic which prints the required features for the intrinsic. Perhaps that is the better way forward.

Reply:
I very much prefer this idea.

Reply:
I agree, I think we would definitely benefit in the long-run if we do not have to modify this code when features are added or changed.

(For future work) is it completely crazy to suggest resolving dependency chains back to either SVE or SME (or both/neither) using `ExtensionDependencies`, which is constructed based on the feature definitions in AArch64Features.td?

Reply:
I suppose you might not need to change arm_sve.td because you could have `SVETargetGuard(x)` imply `sve,(x)`? Likewise for SME.

Reply:
Yes, I believe so.